app/scripts/services/SongPlayer.js

```diff
@@ -28,11 +32,28 @@
             currentSong = song;
         };
 
+        /**
+         * @function playSong
+         * @desc Plays the currentBuzzObject and sets the song object property to true.
+         * @param {Object} song
+         */
+        var playSong = function(song) {
+                currentBuzzObject.play();
+                song.playing = true;
+        };
+
+        /**
+        * @function SongPlayer.play
+        * @desc Takes a song as a parameter and tests to see whether or not the currentBuzzObject
+        is the sames as the current song. If they are different a the new song will load and play.
+        It will also play if they are the same but the song is currently paused.
+        * @param {Object}
+        *
+        */
         SongPlayer.play = function(song) {
             if (currentSong !== song) {
                 setSong(song);
-                currentBuzzObject.play();
-                song.playing = true;
+                playSong(song);
             }
             else if (currentSong === song) {
```

Comment:
minor nit here but prefer doing this:
```js
if (someCondition) {
 //
} else if {
  // other
}
```
avoid putting `else` on a new line because it can trip JS semicolon insertion and cause bugs